### DISCUSSION TITLE
Can't long press with left mouse button when using a tablet for aim

### DISCUSSION THREAD

**Discussion by Shirasagi0012** [ppy/osu#24542] [Q&A]
When using a mouse for clicking and a tablet for aim(the pen tip doesn't touch the tablet surface), I can't complete a slider or spinner with left button. It only strikes once and osu! thinks I released the button when I want to keep holding. However, the right button works fine.

I know it is strange to play like this...
But it gets stranger when I swapped M1 and M2 key binding(lr->rl), the left mouse button still doesn't work. I also tried playing with two different branded mice, one for clicking and another for aim, but under this situation both mouse buttons work fine. I can confirm that both mice and the tablet itself function correctly.

I'm using a Wacom Intuos 5 (PTK-650) tablet. Wacom driver installed but disabled with a script which stops all the Wacom processes and services (not doing so would cause random lag (around 50ms) when using the tablet in osu!lazer, which could be due to a conflict with wacom driver and OpenTabletDrivers)

Logs and demo videos attached. 

[input.log](https://github.com/ppy/osu/files/12337698/input.log)

https://github.com/ppy/osu/assets/23011330/58abf299-c559-42d5-9156-a22de67bdc4b

https://github.com/ppy/osu/assets/23011330/305267f5-b42c-4557-8f70-1d1930a031e6

**Comment by peppy** [comment]
If your tablet is sending mouse buttons this is how it will work. If you use the built-in drivers in lazer you may be able to avoid this.

**Reply by Shirasagi0012** [reply]
I have Wacom's drivers completely disabled, and I can even not control the cursor with the pen in the desktop.  But in osu!lazer the tablet works properly, which should indicate that the built-in drivers have taken effect.

I suspect, when the pen is used for aiming without clicking, the state of the left mouse button being pressed is overwritten with the pen not being clicked due to the continuous updates of the pen's position and button status. This may explain why the key indicators report the left button as pressed for a very short duration, and causes my slider to break :( . 

It appears that osu!lazer have different ways to handle the input from tablet and mouse, and the tablet input has a higher priority than the mouse. This priority can prevent the move of mouse making the cursor flash when using a pen like what Windows does. However, I don't think the pen's tap should have a higher priority than the mouse buttons.
